generate_if_map.c

```diff
@@ -16,26 +16,88 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-#define GATEKEEPER_IF_MAP	"./lua/if_map.lua"
-
 #include <stdio.h>
 #include <string.h>
+#include <argp.h>
 #include <unistd.h>
 #include <ifaddrs.h>
 #include <sys/ioctl.h>
 #include <net/if.h>
 #include <linux/ethtool.h> 
 #include <linux/sockios.h>
 
+/* Argp's global variables. */
+const char *argp_program_version = "generate-if-map 1.0";
+
+/* Arguments. */
+static char adoc[] = "<IF_MAP_LUA>";
+
+static char doc[] = "generate-if-map -- generate the mapping "
+	"network interface device name to its bus information";
+
+
+static struct argp_option options[] = {
+	{"gatekeeper-if-map", 'o', "FILE", 0,
+	"Output the mapping to a Lua FILE"},
```

Comment:
Add `, 1` after the description to enable groups. This is not essential with a single parameter, but it avoids editing that parameter once more parameters are added. Also, indent the second line.